homeassistant/components/duosida_ev/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""The Duosida EV Charger integration."""
+
+from __future__ import annotations
+
+import logging
+
+from duosida_ev import DuosidaCharger
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_DEVICE_ID,
+    CONF_SCAN_INTERVAL,
+    CONF_SWITCH_DEBOUNCE,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_SWITCH_DEBOUNCE,
+)
+from .coordinator import DuosidaDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+    # Future PRs will add:
+    # Platform.BINARY_SENSOR,
+    # Platform.BUTTON,
+    # Platform.NUMBER,
+    # Platform.SWITCH,
+]
+
+type DuosidaConfigEntry = ConfigEntry[DuosidaDataUpdateCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: DuosidaConfigEntry) -> bool:
+    """Set up Duosida EV Charger from a config entry."""
+    host = entry.data[CONF_HOST]
+    port = entry.data.get(CONF_PORT, 9988)
+    device_id = entry.data[CONF_DEVICE_ID]
+
+    scan_interval = entry.options.get(
+        CONF_SCAN_INTERVAL,
+        entry.data.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL),
+    )
```

Comment:
We shouldn't have the scan internal configurable. People can use the update entity service to trigger an update and use their own logic